src/elfpy/pricing_models.py

```diff
@@ -462,132 +472,578 @@ def calc_liquidity(
+            trade_details["init_share_price"],
+            trade_details["share_price"],
+        )
+        (
+            without_fee_or_slippage,
+            output_with_fee,
+            output_without_fee,
+            fee,
+        ) = trade_results
+        market_deltas = {
+            "d_base_asset": trade_details["trade_amount"],
+            "d_token_asset": -output_with_fee,
+            "d_base_asset_slippage": 0,
+            "d_token_asset_slippage": abs(without_fee_or_slippage - output_without_fee),
+            "d_base_asset_fee": 0,
+            "d_token_asset_fee": fee,
+            "d_base_asset_orders": 0,
+            "d_token_asset_orders": 1,
+            "d_base_asset_volume": 0,
+            "d_token_asset_volume": output_with_fee,
+        }
+        wallet_deltas = {
+            "base_in_wallet": -trade_details["trade_amount"],
+            "base_in_protocol": [trade_details["mint_time"], 0],
+            "token_in_wallet": [trade_details["mint_time"], output_with_fee],
+            "token_in_protocol": [trade_details["mint_time"], 0],
+            "fee": [trade_details["mint_time"], fee],
+        }
+        return market_deltas, wallet_deltas
+
+    def close_long(self, trade_details):
+        """
+        take trade spec & turn it into trade details
+        compute wallet update spec with specific details
+            will be conditional on the pricing model
+        """
+        trade_results = self.calc_out_given_in(
+            trade_details["trade_amount"],
+            trade_details["share_reserves"],
+            trade_details["bond_reserves"],
+            trade_details["token_out"],
+            trade_details["fee_percent"],
+            trade_details["stretched_time_remaining"],
+            trade_details["init_share_price"],
+            trade_details["share_price"],
+        )
+        (
+            without_fee_or_slippage,
+            output_with_fee,
+            output_without_fee,
+            fee,
+        ) = trade_results
+        market_deltas = {
+            "d_base_asset": -output_with_fee,
+            "d_token_asset": trade_details["trade_amount"],
+            "d_base_asset_slippage": abs(without_fee_or_slippage - output_without_fee),
+            "d_token_asset_slippage": 0,
+            "d_base_asset_fee": fee,
+            "d_token_asset_fee": 0,
+            "d_base_asset_orders": 1,
+            "d_token_asset_orders": 0,
+            "d_base_asset_volume": output_with_fee,
+            "d_token_asset_volume": 0,
+        }
+        wallet_deltas = {
+            "base_in_wallet": output_with_fee,
+            "base_in_protocol": [trade_details["mint_time"], 0],
+            "token_in_wallet": [trade_details["mint_time"], -1 * trade_details["trade_amount"]],
+            "token_in_protocol": [trade_details["mint_time"], 0],
+            "fee": [trade_details["mint_time"], fee],
+        }
+        return market_deltas, wallet_deltas
```

Comment:
I think it would be nicer to have these functions in the market class. They aren't actually pricing model specific since they just use the generic `calc_in_given_out` and `calc_out_given_in` functions, so we could technically support all of these operations with any pricing model that implements those basic functions. Furthermore, these functions are fairly accounting heavy (especially the short), which isn't really the point of the pricing model from my perspective. We could use the Hyperdrive pricing model to create other actions that aren't enumerated here as well as to accomplish the same goals with different accounting. 

As with the rest of my comments so far, I don't see this as blocking. Having said that, I feel strongly that we should discuss the merits of including this logic in this layer of abstraction.